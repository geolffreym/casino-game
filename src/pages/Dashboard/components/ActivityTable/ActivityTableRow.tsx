import React, { FC } from 'react'
import { TableColumn, TableRow, TableText } from '@components/Table'
import { WithTheme, withTheme } from '@material-ui/core'
import Diamond from '@assets/images/diamond.svg'
import Heart from '@assets/images/heart.svg'
import Spades from '@assets/images/spades.svg'
import Clover from '@assets/images/clover.svg'
import styled from 'styled-components'

/* eslint-disable  @typescript-eslint/explicit-function-return-type */
export interface BalanceTableRowsProps {
  id: string
  result: number[]
  time: string
}

const ActivityTableRowContainer: FC<BalanceTableRowsProps & WithTheme> = ({ id, result, time, theme }): JSX.Element => {
  // @ts-expect-error
  const color = theme.palette.grey[250]
  const getIcon = (icon: number) => {
    switch (icon) {
      case 1: return <Spades />
      case 2: return <Clover />
      case 3: return <Diamond />
      case 4: return <Heart />
      default: return <span>?</span>
    }
  }

  return (
    <TableRow>
      <TableColumn>
        <TableText color={color}>
          {id}
        </TableText>
      </TableColumn>
      <TableColumn>
        <ResultWrapper>
          {getIcon(result[0])}
          {getIcon(result[1])}
          {getIcon(result[2])}
        </ResultWrapper>
      </TableColumn>
      <TableColumn>
        {time}
      </TableColumn>
    </TableRow>
  )
}

const ResultWrapper = withTheme(
  styled.div<WithTheme>`
      display: flex;
      
      svg {
        width: 1rem;
        height: 1rem;
        margin-right: 0.5rem;
        fill: ${({ theme }) => theme.palette.grey[250]}
      }
    `
)

export const ActivityTableRow = withTheme(ActivityTableRowContainer)
