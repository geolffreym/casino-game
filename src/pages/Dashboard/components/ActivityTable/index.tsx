export { ActivityTable as default } from '@pages/Dashboard/components/ActivityTable/ActivityTableContainer'
