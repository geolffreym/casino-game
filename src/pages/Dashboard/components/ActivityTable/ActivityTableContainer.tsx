import React, { FC, useState } from 'react'
import { Theme } from '@material-ui/core'
import { useTheme } from '@material-ui/core/styles'
import { TablePagination, TableWrapper } from '@components/Table'
import ActivityTableView from '@pages/Dashboard/components/ActivityTable/ActivityTableView'
import { connect, RootStateOrAny } from 'react-redux'
import { Activity, ActivityState } from '@state/reducers/activityReducers'

/* eslint-disable  @typescript-eslint/explicit-function-return-type */

const ActivityTableContainer: FC<ActivityState> = ({ activity }): JSX.Element => {
  const theme: Theme = useTheme()
  const [page, setPage] = useState(0)
  const [rowsPerPage, setRowsPerPage] = useState(10)
  const slice: number[] = [page * rowsPerPage, (page * rowsPerPage) + rowsPerPage]

  const onPageChange = (event: unknown, newPage: number) => {
    setPage(newPage)
  }

  const onRowsPerPageChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setPage(0)
    setRowsPerPage(parseInt(event.target.value, 10))
  }

  const columns: any[] = [
    { id: 'id', name: 'ID' },
    { id: 'result', name: 'Result' },
    { id: 'time', name: 'Time' }
  ]

  const rows: Activity[] = activity

  return (
    <TableWrapper theme={theme}>
      <ActivityTableView {...{ columns, rows, slice, theme }} />
      <TablePagination
        count={rows.length}
        rowsPerPageOptions={[10]}
        component='div'
        {...{
          onPageChange,
          onRowsPerPageChange,
          rowsPerPage,
          page
        }}
      />
    </TableWrapper>
  )
}

const mapStateToProps = (state: RootStateOrAny): ActivityState => {
  return {
    activity: state.activity.activity
  }
}

export const ActivityTable = connect(
  mapStateToProps
)(ActivityTableContainer)
