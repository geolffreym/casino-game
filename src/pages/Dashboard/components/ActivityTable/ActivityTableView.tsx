import React, { FC } from 'react'
import Table, {
  TableBody, TableColumn, TableHeader,
  TableScroller, TableRow, TableRowEmpty
} from '@components/Table'
import { ActivityTableRow } from '@pages/Dashboard/components/ActivityTable/ActivityTableRow'

/* eslint-disable  @typescript-eslint/explicit-function-return-type */
export interface BalanceTableViewProps {
  rows: any[]
  columns: any[]
  slice: number[]
}

const ActivityTableView: FC<BalanceTableViewProps> = ({ columns, rows, slice }): JSX.Element => {
  return (
    <TableScroller>
      <Table stickyHeader aria-label='Balance table'>
        <TableHeader>
          <TableRow>
            {
              columns.map((column, i) =>
                <TableColumn key={i}>
                  {column.name}
                </TableColumn>
              )
            }
          </TableRow>
        </TableHeader>
        <TableBody>
          {
            (rows.slice(...slice).length > 0)
              ? rows.slice(...slice).map((row, i) => {
                return <ActivityTableRow {...row} key={i} />
              })
              : <TableRowEmpty />
          }
        </TableBody>
      </Table>
    </TableScroller>
  )
}

export default ActivityTableView
