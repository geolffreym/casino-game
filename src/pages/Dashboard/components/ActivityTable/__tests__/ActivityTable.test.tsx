import ActivityTable from '@pages/Dashboard/components/ActivityTable/index'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import React from 'react'
import { createMemoryHistory } from 'history'
import createStore from '@state/store'
import { Provider } from 'react-redux'

/* eslint-disable no-undef */
describe('<ActivityTable />', () => {
  it('should render', () => {
    const history = createMemoryHistory()
    const store = createStore(history)
    const component = shallow(
      <Provider store={store}>
        <ActivityTable />;
      </Provider>
    )
    expect(component).toMatchSnapshot()
  })
})
