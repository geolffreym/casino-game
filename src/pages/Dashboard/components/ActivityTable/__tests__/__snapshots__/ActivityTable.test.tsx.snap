// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActivityTable /> should render 1`] = `ShallowWrapper {}`;
